src/command.rs

```diff
@@ -54,9 +54,14 @@ impl Command {
       Self::Operation(operation) => state.operation = *operation,
       Self::Print => state.print()?,
       Self::Repl => {
-        for result in BufReader::new(io::stdin()).lines() {
-          let line = result?;
-          match line.trim().parse::<Self>() {
+        let history = home_dir().unwrap_or_default().join(".degenerate_history");
+
+        let mut editor = Editor::<()>::new();
+        editor.load_history(&history).ok();
+
+        while let Ok(line) = editor.readline("> ") {
```

Comment:
I just realized that we should probably exit if `editor.readline()` returns an error. If the user does `ctrl-c` for example, we should probably terminate and not run further commands.

```suggestion
        let line = editor.readline("> ")?
```

Reply:
Do we want to print out the error message (do what's in main already) or exit quietly?

Reply:
Ahh, right. If it's `Eof` (ctrl-d) or `Enterrupted` (ctrl-c) we don't want to print the error message and shouldn't return an error code. Normally we would just check which error variant it is, but that's not possible because we're using `Box<dyn Error>`. However, `Box<dyn Error>` has a [downcast](https://doc.rust-lang.org/std/error/trait.Error.html#method.downcast) method, so we can try to downcast to `rustyline::error::ReadlineError` in main and check which variant it is.